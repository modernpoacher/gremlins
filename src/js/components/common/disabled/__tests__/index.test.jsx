import React from 'react'
import {
  snapshotOf
} from 'react-component-snapshot'

import '@testing-library/jest-dom'

import {
  render
} from '@testing-library/react'

import Disabled from '#gremlins/components/common/disabled'

describe('#gremlins/components/common/disabled', () => {
  describe('<Disabled />', () => {
    it('renders', () => {
      const {
        container: {
          firstElementChild: disabled
        }
      } = render(
        <Disabled />
      )

      expect(disabled)
        .toBeInstanceOf(HTMLSpanElement)
    })

    /**
     *  @deprecated For migration toward Testing Library
     */
    it('matches the snapshot', () => {
      const {
        container: {
          firstElementChild: disabled
        }
      } = render(
        <Disabled />
      )

      expect(snapshotOf(disabled))
        .toMatchSnapshot()
    })
  })
})
