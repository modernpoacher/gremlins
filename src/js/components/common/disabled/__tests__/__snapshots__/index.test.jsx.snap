// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#gremlins/components/common/disabled <Disabled /> matches the snapshot 1`] = `
<span
  className="is-disabled"
>
  *
</span>
`;
